startup/10-devices.py

```diff
@@ -183,20 +183,25 @@ def _update_readback_setpoint(self):
 
     def put(self, value):
         self._setpoint = float(value)
-
+        
         start_time = ttime.monotonic()
         getattr(atf_db, f"put_{self._dtype}")(
             atf_db.get_channel_index(
                 f"{self._db}::{self._psname};{self._write_suffix}"
             ),
             value,
         )
-        while ttime.monotonic() - start_time < self._timeout:
-            if abs(self._get_readback() - value) < self._tol:
-                break
-            else:
-                # not reached yet, wait a bit
-                ttime.sleep(0.1)
+        try:
+            if(ttime.monotonic() - start_time >= self._timeout):
+                raise Exception("Timeout")
+            while ttime.monotonic() - start_time < self._timeout:
+                if abs(self._get_readback() - value) < self._tol:
+                    break
+                else:
+                    # not reached yet, wait a bit
+                    ttime.sleep(0.1)
+        except Exception as err:
```

Comment:
On this line we can `raise TimeoutException(message)`.

In the `message`, I suggest to include the timeout value we set for the signal (it can be different for different signals) and also the `self.name`, and maybe some other parameters if it helps to identify the specific element that is failing (read/write suffix, maybe tolerance, etc.)